### DISCUSSION TITLE
Как построить REST API с помощью JS, Node.js и Express.js ⚡️ Node.js с примерами кода

### DISCUSSION THREAD

**Discussion by giscus** [xsltdev/nodejsdev.ru#52] [Announcements]
# Как построить REST API с помощью JS, Node.js и Express.js ⚡️ Node.js с примерами кода

статью, которая легко усваивается, и привел практический пример, которому можно следовать. В итоге мы создадим полноценный API, применяя одну лучшую практику за другой

https://nodejsdev.ru/guides/rest-api-design/

**Comment by seyton35** [comment]
привет, отличная статья, узнал много нового
есть ошибка, пропущена реализация recordsController.js
вот мой вариант, по примеру вашего
<b>
const recordService = require('../services/recordService')

const getRecordsForWorkout = (req, res) => {
    const { params: { workoutId } } = req
    if (!workoutId) res.status(400).send({
        status: 'Failed',
        data: {
            message: "Parameter 'workoutId' can not be empty"
        }
    })
    try {
        const allRecordsForWorkout = recordService.getRecordForWorkout(workoutId)
        res.send({ status: 'Ok', data: allRecordsForWorkout })
    } catch (error) {
        res.status(error?.status || 500).send({
            status: 'Failed',
            data: {
                error: error?.message || error
            }
        })
    }
}

module.exports = {
    getRecordsForWorkout,
}
</b>
